modules/python/src2/cv2.cpp

```diff
@@ -206,6 +206,23 @@ bool pyopencv_to(PyObject* obj, T& p, const char* name = "<unknown>");
 template<typename T> static
 PyObject* pyopencv_from(const T& src);
 
+template <typename T> static
+bool pyopencv_to_ptr(PyObject *o, T* p, const char *name = "<unknown>")
+{
+    if (!o || o == Py_None)
+        return true;
+    p = new T();
```

Comment:
@mshabunin Seems you were not convinced by my reply, so I have created two extra commits to make things concrete.

The first two commits fixes the Python and Java wrappers, and the last tests all wrappers using `cv::EMD` as an example, in which we failed to wrap directly for quite long time.